### PR TITLE
add expiryDetail to google_cloud_identity_group_membership

### PR DESCRIPTION
**Pull request by sqin2019**
fixes https://github.com/hashicorp/terraform-provider-google/issues/10343

Release Note Template for Downstream PRs (will be copied)
```release-note:enhancement
cloudidentity: added `expiry_detail` field to `google_cloud_identity_group_membership` resource
```

### CODE REVIEW

**Comment by shuyama1** [comment]
@sqin2019 please ignore the missing test report in this PR. I believe the tests for this resource is set up to run serial and our missing test detector seems have issue with this test setup

File: mmv1/templates/terraform/examples/cloud_identity_group_membership_user.tf.erb [hunk 1]
```diff
@@ -21,6 +21,9 @@ resource "google_cloud_identity_group_membership" "cloud_identity_group_membersh
 
   roles {
     name = "MEMBER"
+    expiry_detail = {
```
Comment (shuyama1) [inline mmv1/templates/terraform/examples/cloud_identity_group_membership_user.tf.erb:24]:
```suggestion
    expiry_detail {
```

Comment (sqin2019) [reply mmv1/templates/terraform/examples/cloud_identity_group_membership_user.tf.erb]: I see, thanks!

**Comment by sqin2019** [comment]
@shuyama1 Hi, the test failed with below error, same perferred_member_key used in other test cases passed, and I didn't make any changes to this field, would you be interested in taking a look? Thanks!
```
          {
            "@type": "type.googleapis.com/google.rpc.BadRequest",
            "fieldViolations": [
              {
                "description": "Request contains invalid argument(s).",
                "field": "resource.preferred_member_key"
              }
            ]
          }
```

**Comment by shuyama1** [comment]
> @shuyama1 Hi, the test failed with below error, same perferred_member_key used in other test cases passed, and I didn't make any changes to this field, would you be interested in taking a look? Thanks!
> 
> ```
>           {
>             "@type": "type.googleapis.com/google.rpc.BadRequest",
>             "fieldViolations": [
>               {
>                 "description": "Request contains invalid argument(s).",
>                 "field": "resource.preferred_member_key"
>               }
>             ]
>           }
> ```

Sure, I'll take a look at the test failure later today

File: mmv1/templates/terraform/examples/cloud_identity_group_membership.tf.erb [hunk 2]
```diff
@@ -35,5 +35,8 @@ resource "google_cloud_identity_group_membership" "<%= ctx[:primary_resource_id]
 
   roles {
   	name = "MEMBER"
```
Comment (sqin2019) [inline mmv1/templates/terraform/examples/cloud_identity_group_membership.tf.erb:37]: note: fix indentation

**Comment by sqin2019** [comment]
> debug log



> > @shuyama1 Hi, the test failed with below error, same perferred_member_key used in other test cases passed, and I didn't make any changes to this field, would you be interested in taking a look? Thanks!
> > ```
> >           {
> >             "@type": "type.googleapis.com/google.rpc.BadRequest",
> >             "fieldViolations": [
> >               {
> >                 "description": "Request contains invalid argument(s).",
> >                 "field": "resource.preferred_member_key"
> >               }
> >             ]
> >           }
> > ```
> 
> Sure, I'll take a look at the test failure later today

Adding screenshot, seems with expiry it will have the invalid preferredMemberKey argument, but I am not sure why.
Screenshot 2023-11-10 at 2 39 26 PM
Screenshot 2023-11-10 at 2 39 07 PM

**Comment by shuyama1** [comment]
Worked with internal team to get it upgrade to the premium plan. Trigger a rerun to see if it works.

/gcbrun

**Comment by sqin2019** [comment]
@shuyama1 Hi, I am done with my change, please take a look.

**Comment by shuyama1** [comment]
> @shuyama1 Hi, I am done with my change, please take a look.

Thanks! will take another look today

**Review by shuyama1 — COMMENTED** [review]
I think the update should work in most of the cases, but I wonder if the following case will run into issues:
update from:
```
roles {
    name = "MEMBER"
}
roles {
    name = "MANAGER"
}
```
to
```
roles {
    name = "MEMBER"
    expiry_detail {
      expire_time = "xxx"
    }
}
```
or vice versa. Basically, it will require to update + remove/add roles but based on https://cloud.google.com/identity/docs/reference/rest/v1/groups.memberships/modifyMembershipRoles, it seems an invalid request. 

If it's not allowed, we may want to use custom post_update instead to separate calls for adding/removing and updating.

**Comment by sqin2019** [comment]
@shuyama1 updated to use custom_update, please take a look, thx!

**Review by shuyama1 — COMMENTED** [review]
LGTM! Thanks. Triggering a new run to before merging it just incase

**Comment by shuyama1** [comment]
The failing test is not related and I've already created a fix for it.

**Review by shuyama1 — APPROVED** [review]

**Pull request merged**